### PR TITLE
Add support for query param in Projects list API

### PR DESCRIPTION
**Pull request by mteodori**
The new Projects API list endpoint provides a `q` parameter to query projects by name: https://developer.hashicorp.com/terraform/cloud-docs/api-docs/projects#list-projects

This PR add supports for it leveraging what is already available in the parent class

### CODE REVIEW

File: terrasnek/projects.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def list(self, page=None, page_size=None, filters=None):
             <https://developer.hashicorp.com/terraform/cloud-docs/api-docs/projects#query-parameters>`__
         """
         return self._list(\
-            self._org_api_v2_base_url, page=page, page_size=page_size, filters=filters)
+            self._org_api_v2_base_url, query=None, filters=filters, page=page, page_size=page_size)
```
Comment (dahlke) [inline terrasnek/projects.py:62]: You want `query=query` here, otherwise I'll approve this.

**Comment by dahlke** [comment]
I just added this functionality to the main branch, will close this out once I cut a new release.

File: terrasnek/projects.py (hunk 1 above)
Comment (mteodori) [reply terrasnek/projects.py]: I've seen you've already applied, thank you!

**Comment by dahlke** [comment]
0.1.12 has been released which includes this fix.

**Pull request closed without merging**